From 5fc054beb415344a0ce44c19e554937c9158d08e Mon Sep 17 00:00:00 2001
From: Jim Meyering <meyering@redhat.com>
Date: Tue, 9 Oct 2012 14:02:15 +0200
Subject: [PATCH 34/89] tests: improve test for partitionable loop devices

* tests/init.cfg (require_partitionable_loop_device_): Skip
when cat fails.
---
 tests/init.cfg | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/tests/init.cfg b/tests/init.cfg
index 24b10bc..dc8b2bc 100644
--- a/tests/init.cfg
+++ b/tests/init.cfg
@@ -120,8 +120,9 @@ require_erasable_()
 # At least Fedora 16 (kernel 3.1.6-1.fc16.x86_64) fails this test.
 require_partitionable_loop_device_()
 {
-  case $(cat /sys/devices/virtual/block/$(basename $1)/ext_range) in
-    0|1) skip_ your system does not support loop partitioning;;
+  local f=/sys/devices/virtual/block/$(basename $1)/ext_range
+  case $(cat "$f") in
+    ''|0|1) skip_ your system is not configured to partition loop devices;;
   esac
 }
 
-- 
1.8.5.3

